pkg/labels/selector_test.go

```diff
@@ -306,65 +501,97 @@ func TestRequirementLabelSelectorMatching(t *testing.T) {
+		}}, true, true},
+		{"0.1.2.domain/99 notin (10.10.100.1, tick.tack.clock)", &LabelSelector{Requirements: []Requirement{
+			getRequirement("0.1.2.domain/99", NotIn, util.NewStringSet("10.10.100.1", "tick.tack.clock"), SetBasedMatch, t),
 		}}, true, true},
 		{"foo  in	 (abc)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("foo", In, util.NewStringSet("abc"), t),
+			getRequirement("foo", In, util.NewStringSet("abc"), SetBasedMatch, t),
 		}}, true, true},
-		{"x not\n\tin (abc)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", NotIn, util.NewStringSet("abc"), t),
+		{"x notin\n (abc)", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", NotIn, util.NewStringSet("abc"), SetBasedMatch, t),
 		}}, true, true},
-		{"x  not in	\t	(abc,def)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", NotIn, util.NewStringSet("abc", "def"), t),
+		{"x  notin	\t	(abc,def)", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", NotIn, util.NewStringSet("abc", "def"), SetBasedMatch, t),
 		}}, true, true},
 		{"x in (abc,def)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet("abc", "def"), t),
+			getRequirement("x", In, util.NewStringSet("abc", "def"), SetBasedMatch, t),
 		}}, true, true},
 		{"x in (abc,)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet("abc", ""), t),
+			getRequirement("x", In, util.NewStringSet("abc", ""), SetBasedMatch, t),
 		}}, true, true},
 		{"x in ()", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", In, util.NewStringSet(""), t),
+			getRequirement("x", In, util.NewStringSet(""), SetBasedMatch, t),
 		}}, true, true},
-		{"x not in (abc,,def),bar,z in (),w", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", NotIn, util.NewStringSet("abc", "", "def"), t),
-			getRequirement("bar", Exists, nil, t),
-			getRequirement("z", In, util.NewStringSet(""), t),
-			getRequirement("w", Exists, nil, t),
+		{"x notin (abc,,def),bar,z in (),w", &LabelSelector{Requirements: []Requirement{
+			getRequirement("bar", Exists, nil, SetBasedMatch, t),
+			getRequirement("w", Exists, nil, SetBasedMatch, t),
+			getRequirement("x", NotIn, util.NewStringSet("abc", "", "def"), SetBasedMatch, t),
+			getRequirement("z", In, util.NewStringSet(""), SetBasedMatch, t),
 		}}, true, true},
 		{"x,y in (a)", &LabelSelector{Requirements: []Requirement{
-			getRequirement("y", In, util.NewStringSet("a"), t),
-			getRequirement("x", Exists, nil, t),
+			getRequirement("y", In, util.NewStringSet("a"), SetBasedMatch, t),
+			getRequirement("x", Exists, nil, SetBasedMatch, t),
 		}}, false, true},
+		{"x=a", &LabelSelector{Requirements: []Requirement{
+			getRequirement("x", In, util.NewStringSet("a"), ExactMatch, t),
+		}}, true, true},
+		{"x=a,y!=b", &LabelSelector{Requirements: []Requirement{
```

Comment:
I would like to see a test that combines the exact-match syntax and the set-based syntax, such as:
```
x=a,y!=b,z in (h,i,j)
```